File: version_3/version_3/Hr_manager.cpp
/***********************************************************************
* Module:  Hr_manager.cpp
* Author:  Anton
* Modified: Monday, November 7, 2016 6:57:48 PM
* Purpose: Implementation of the class Hr_manager
***********************************************************************/
#include <vector>
#include <string>
#include "Candidate.h"
#include "Hr_manager.h"
#include "Technical_manager.h"

//constructors
Hr_manager::Hr_manager():
	name("Name")
{
	//std::cout << "HR-manager was created (default)" << std::endl;
}

Hr_manager::Hr_manager(std::string name) :
	name(name)
{
	//std::cout << "HR-manager was created (inicialisation)" << std::endl;
}

Hr_manager::Hr_manager(Hr_manager &sHr_manager):
	name(sHr_manager.name)
{
	//std::cout << "HR-manager was created (copy)" << std::endl;
}

//destructor
Hr_manager::~Hr_manager()
{
	//std::cout << "HR- manager was destroyed" << std::endl;
	//std::cout << std::endl;
}
////////////////////////////////////////////////////////////////////////
// Name:       Hr_manager::prepareVacancyList()
// Purpose:    Implementation of Hr_manager::prepareVacancyList()
// Return:     void
////////////////////////////////////////////////////////////////////////

std::vector<std::string> Hr_manager::prepareVacancyList(int max, int cur)//can be better
{
	std::vector<std::string> list;
	for (int i = 0; i < (max - cur); i++)
	{
		std::string vac;
		std::cin >> vac;
		list.push_back(vac);
	}
	return list;
	// TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Hr_manager::chooseCandidates()
// Purpose:    Implementation of Hr_manager::chooseCandidates()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Hr_manager::chooseCandidates(std::vector<std::string> list)
{
	for (int i = 0; i < candidate.size(); i++)
	{
		for (int j = 0; j < list.size(); j++)
		{
			if (candidate[i]->resume.getCompetence() == list[j])
			{
				candidate[i]->setState("review");
				break;
			}
		}
	}
	// TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Hr_manager::get()
// Purpose:    Implementation of Hr_manager::get()
// Return:     void
////////////////////////////////////////////////////////////////////////

//getters
std::string  Hr_manager::getName(void)
{
	return name;
}

////////////////////////////////////////////////////////////////////////
// Name:       Hr_manager::set()
// Purpose:    Implementation of Hr_manager::set()
// Return:     void
////////////////////////////////////////////////////////////////////////

//setters
void Hr_manager::setName(std::string n)
{
	name = n;
}

File: cs_version4/cs_version4/Hr_manager.cs
/***********************************************************************
 * Module:  Hr_manager.cs
 * Author:  Anton
 * Purpose: Definition of the Class Hr_manager
 ***********************************************************************/

using System;
using System.Collections.Generic;

public class Hr_manager : Worker
{
    public Hr_manager() : base() 
    {
        //name = "n";
        name = "dddd";
        Console.WriteLine("HR-manager was created (default)");
    }
	public Hr_manager(string n) : base(n)
    {
        //name = n;
        Console.WriteLine("HR-manager was created (inicialisation)");
    }
	public Hr_manager(Hr_manager sHr_manager)
    {
        //name = sHr_manager.name;
        Console.WriteLine("HR-manager was created (copy)");
    }
   public List<string> prepareVacancyList(int max, int cur)
    {
       List<string> list = new List<string>(1);
	for (int i = 0; i < (max - cur); i++)
	{
		string vac;
		vac = Console.ReadLine();
		list.Add(vac);
	}
	return list;
    }

	public void chooseCandidates(List<string> list)
   {
       for (int i = 0; i < candidate.Count; i++)
       {
           for (int j = 0; j < list.Count; j++)
           {
               if (candidate[i].resume.getCompetence() == list[j])
               {
                   candidate[i].setState("review");
                   break;
               }
           }
       }
   }

   public List<Candidate> candidate = new List<Candidate>(1);
    private string name;
    public  static string operator + (Hr_manager left, string n)
	{
        return left.name + n;
	}
    public static string operator -(Hr_manager left, int n)
    {
        for (int i = 0; i < n; i++)
        {
            if (left.name.Length > 0)
            {
                left.name.Remove(left.name.Length - 1);
            }
        }
        return left.name;
    }
}